src/browser/components/TabNavigation/Navigation.tsx

```diff
@@ -94,37 +87,44 @@ class Navigation extends Component<NavigationProps, NavigationState> {
     prevProps: NavigationProps,
     prevState: NavigationState
   ): void {
-    if (prevProps.openDrawer !== this.props.openDrawer) {
-      const newState: NavigationState = {}
-      if (this.props.openDrawer) {
-        newState.drawerContent = this.props.openDrawer
+    let lastSelectedDrawerName: string | null = null
+    let newTransitionState: DrawerTransitionState = prevState.transitionState
+    if (prevProps.selectedDrawerName !== this.props.selectedDrawerName) {
+      if (this.props.selectedDrawerName) {
         if (
           prevState.transitionState === Closed ||
           prevState.transitionState === Closing
         ) {
-          newState.transitionState = Opening
+          newTransitionState = Opening
+          lastSelectedDrawerName = null
```

Comment:
Personal preference but I find this easier to read as two setState calls rather than creating and mutating local variables and then only calling once